airflow/providers/amazon/aws/hooks/athena.py

```diff
@@ -228,7 +228,7 @@ def poll_query_status(self, query_execution_id: str, max_tries: Optional[int] =
             sleep(self.sleep_time)
         return final_query_state
 
-    def get_output_location(self, query_execution_id: str) -> str:
+    def get_output_location(self, query_execution_id: str) -> Optional[str]:
```

Comment:
This should not be optional, if it does not return, it will raise. So `str` is correct

Reply:
The reason its Optional is because it can return None as output_location can be None

`        output_location = None
`
Fixes this mypy error
```airflow/providers/amazon/aws/hooks/athena.py:253: error: Incompatible return value type (got "Optional[Any]", expected "str")
            return output_location```

Reply:
In the following block, the only way the `output_location` can be `None` is if `response` is a false-y value which does not look like it is possible from https://boto3.amazonaws.com/v1/documentation/api/1.9.42/reference/services/athena.html#Athena.Client.get_query_execution and https://docs.aws.amazon.com/athena/latest/APIReference/API_GetQueryExecution.html

https://github.com/apache/airflow/blob/d9259a556e9ae4c60edad9d1374ca0c7a9d2fa58/airflow/providers/amazon/aws/hooks/athena.py#L240-L251

Reply:
Correct, We just have to cover the `else` for the `if response` to fix the mypy error.

```airflow/providers/amazon/aws/hooks/athena.py:253: error: Incompatible return value type (got "Optional[Any]", expected "str")
            return output_location```

Reply:
You can raise an error in the else part

Reply:
Thanks, changed.